terps/magnetic/Generic/defs.h

```diff
@@ -35,7 +35,7 @@
 *       correct number of bits on your system !!!
 \*****************************************************************************/
 
-#ifdef GARGLK
+#ifdef MAGNETIC_GLKUNIX

```

Comment:
OK, so there's a wrinkle here... `GARGLK` really means _three_ things, unfortunately:

1. Unix Glk startup (fixed in this request)
2. Running under Gargoyle
3. Changes from upstream

The only thing it _should_ be used for is 2. But because of 3, if you don't define `GARGLK`, then you'll get bad code (e.g. loads of pointer sign comparison warnings, which are technically invalid code even if most/all compilers do the "expected" thing; plus possibly passing the wrong types to sprintf in one instance).

I made these changes to Magnetic _before_ I started the whole upstream branch thing, so I used `#ifdef GARGLK` to indicate my changes.  Given how things work now, it's better to just make the changes, and use `git diff` to see the differences.

So what I'd ask you to do is this: find all `GARGLK`-wrapped code which is clearly a bug fix.  To my recollection this is a call to sprintf() in the Glk code, and various casts/type-related stuff in the Glk code as well as the generic code.

Just completely get rid of the code that isn't protected by the macro.  For example:

```
#ifdef GARGLK
type8s* hint_contents = 0;
#else
type8* hint_contents = 0;
#endif
```
becomes
```
type8s* hint_contents = 0;
```
And do that here in `defs.h`, too: let's just rip out the old typedefs.  `stdint.h` is from C99, so we can unconditionally use it. If somebody is still using a C89 compiler, it's on _them_ to get this working.

There is a bunch of graphics-related Gargoyle changes that can remain wrapped in `GARGLK` for now.  I haven't decided how to deal with that; but we really should just have bug fixes in the code here, unconditionally.

Reply:
«No so fast Joan» :)
Thanks for taking the time to explain. I'll take a look into it, but it has to be in a couple of days (I go on a trip).

Reply:
If I hope I did this correctly and should compile for Garglk.